### PR TITLE
nix: clean up fwupd-efi handling a little bit

### PR DESCRIPTION
**Pull request by lilyinstarlight**
Since there were open comments still at https://github.com/nix-community/lanzaboote/pull/131#pullrequestreview-1345125430 when #131 was merged, I'm opening this PR to allow further feedback from @dasJ on the proposed diff whenever they get a chance (no rush though!)

Primarily it removes the `mkdir` call in favor of letting systemd handle the `/run` directory lifetime and symlinks in `fwupdx64.efi` instead of copying it (since linking works just fine)

### CODE REVIEW

**Comment by RaitoBezarius** [comment]
Oh sorry, I misinterpreted the discussion as closed somehow. I apologize to @dasJ and you.

**Comment by lilyinstarlight** [comment]
> Oh sorry, I misinterpreted the discussion as closed somehow. I apologize to @dasJ and you.

No worries! It was in a working state and having fwupd support is better than not, so merging wasn't really a problem (else I would have made sure to mark it draft)

File: nix/modules/lanzaboote.nix [hunk 1]
```diff
@@ -117,10 +117,23 @@ in
...
+        Type = "oneshot";
+        RemainAfterExit = true;
+        RuntimeDirectory = "fwupd-efi";
+      };
       # Place the fwupd efi files in /run and sign them
-      preStart = ''
-        mkdir -p /run/fwupd-efi
-        cp ${config.services.fwupd.package.fwupd-efi}/libexec/fwupd/efi/fwupd*.efi /run/fwupd-efi/
+      script = ''
+        ln -s ${config.services.fwupd.package.fwupd-efi}/libexec/fwupd/efi/fwupd*.efi /run/fwupd-efi/
```
Comment (dasJ) [inline nix/modules/lanzaboote.nix:136]: Wouldn't `ln -fs` make more sense so I can restart the service multiple times?

**Review by blitz — DISMISSED** [review]
Thanks for the PR! Can you address @dasJ's comment?

File: nix/modules/lanzaboote.nix (hunk 1 above)
Comment (lilyinstarlight) [reply nix/modules/lanzaboote.nix]: Done!

**Review by RaitoBezarius — APPROVED** [review]

**Pull request merged**